File: src/main/java/com/example/summarizer/enums/SummaryLength.java
package com.example.summarizer.enums;

public enum SummaryLength {
    TLDR("1-2 sentences", 0.1),
    SHORT("3-5 sentences", 0.2),
    MEDIUM("5-10 sentences", 0.4),
    DETAILED("Comprehensive", 0.6);

    private final String description;
    private final double targetRatio;

    SummaryLength(String description, double targetRatio) {
        this.description = description;
        this.targetRatio = targetRatio;
    }

    public String getDescription() { return description; }
    public double getTargetRatio() { return targetRatio; }
}

File: summarizer-ui/src/enum/summary-length.ts
export type SummaryLength = "TLDR" | "SHORT" | "MEDIUM" | "DETAILED";

export const SummaryLengthMeta = {
    TLDR: {
        description: "1-2 sentences",
        targetRatio: 0.1,
    },
    SHORT: {
        description: "3-5 sentences",
        targetRatio: 0.2,
    },
    MEDIUM: {
        description: "5-10 sentences",
        targetRatio: 0.4,
    },
    DETAILED: {
        description: "Comprehensive",
        targetRatio: 0.6,
    },
} as const;